lib/ClangImporter/ImportMacro.cpp

```diff
@@ -478,6 +573,12 @@ static ValueDecl *importMacro(ClangImporter::Implementation &impl,
       firstValue     = firstInt->first;
       firstSwiftType = firstInt->second;
     } else {
+      emitMacroDiagnostic(impl,
+                          Diagnostic(diag::macro_not_imported, name.str()),
+                          macro->getDefinitionLoc(), ClangN);
+      emitMacroDiagnostic(
+          impl, Diagnostic(diag::macro_not_imported_unsupported_operand),
+          tokenI[0].getLocation(), ClangN);
```

Comment:
I think this is "too early" to call this "an operand".

If I understand correctly, this is done for every 3 token macros. In the previous added block you rejected the `<identifier><l_paren><r_paren>` case, but in here, if you don't find an integer constant, you call it "an operand", even if it can be dozens of other things (for example: `#define MY_MACRO { 1 }`).

I think in this case you can only invoke the `macro_not_imported_unsupported_structure` because you are looking for two numbers in positions 0 and 2, and you don't find them. Nothing says this was an "operand" of the "good path"

Reply:
👍